### DISCUSSION TITLE
Best Practices for Modular monolith

### DISCUSSION THREAD

**Discussion by wafotey** [dotnet/eShop#263] [General]
Microsoft is focus on microservices with it product and the eshoponcontainers project is fine for that. However can we have a corresponding modular monolith on this project with best practices from the industry. Since modular monolith has become the go to for most people in the industries to start with. Also there seem to be less books and material on this topic compared with microservices.

**Comment by ks1990cn** [comment]
.Net API template is there to start with your monolith. Just you can use some patterns to create good folder structure, like data access layer or repository patterns. Any other manager layer for entrypoint of your services. Your engine layer where you can write your logic.

**Comment by jamesmontemagno** [comment]
We do have many eBooks around the topic https://dotnet.microsoft.com/en-us/learn/dotnet/architecture-guides 

ALso: https://learn.microsoft.com/en-us/azure/architecture/web-apps/guides/reliable-web-app/overview

we are also working on a new ebook for this sample as well as .NET Aspire continues to evolve.
